sktime/classification/early_classification/base.py

```diff
@@ -353,7 +352,6 @@ def split_indices_and_filter(X, indices, decisions):
         inv_dec = np.invert(decisions)
         return X[inv_dec], indices[inv_dec], indices[decisions]
 
-    @abstractmethod
```

Comment:
So - I suppose this may have already come up in the original depreciation discussion, in which case if consensus was reached can disregard my comment, but I personally really like using abstract methods.  Reasoning - for developers if forces you to actually write the `_fit` and `_predict` functions.  I think for anyone unfamiliar with the templating design pattern, the fact that the code literally breaks if you don't write a `_fit` makes it easy to figure out that that is where the logic should live. 

I understand that there are also downsides to requiring `abc` though, but just wanted to put that out there - for the type of design we do it is a pretty nice tool.

Reply:
for the PR at hand: it breaks the deprecation action, but since I removed the deprecation action (was broken/incomplete), we do not need to remove `abc` in here. I erroneously only reverted the estimator file, not the base class file. I will revert the base class file.

for the general design point: yes, I do appreciate the clarity `abc` brings, but the drawback (which I've already encountered) is that inspection becomes much more difficult due to the added decorators (these are decorators, so you may have to unwrap to inspect!)
this has caused problems in inspection heavy use cases such as `reset` like methods (in `BaseObject` work), or work around the polymorphic base class type, although it is currently not breaking anything on `main`.

There is an open issue on the general design point if you want to chip in: https://github.com/sktime/sktime/issues/1923